helm-chart/renku/values.yaml

```diff
@@ -116,14 +118,55 @@ ingress:
 
 ## Keycloak configuration
 keycloak:
+
+  ## Spawn a keycloak instance
+  enabled: true
+
+  postgresql:
+    # Disable PostgreSQL dependency
+    enabled: false
+
+  extraEnv: |
+    - name: DB_VENDOR
+      value: postgres
+    - name: DB_PORT
+      value: "5432"
+    - name:  PROXY_ADDRESS_FORWARDING
+      value: "true"
+
+
+  # The following environment variables are provided to keycloak
+  # as extraEnvFrom secrets.
+
+  # renku-keycloak-postgres
+  # - DB_ADDR: database host                            value: <RELEASE-NAME>-postgresql
+  # - DB_DATABASE: database name for keycloak           value: global.keycloak.postgresDatabase
+  # - DB_USER: database username for keycloak database  value: global.keycloak.postgresUser
+  # - DB_PASSWORD: database password for keycloak       value: global.keycloak.postgresPassword.value
+  # keycloak-password-secret
+  # - KEYCLOAK_USER: keyclaok admin username            value: global.keycloak.user
+  # - KEYCLOAK_PASSWORD: keycloak admin password        value: global.keycloak.password.value
+
+  extraEnvFrom: |
```

Comment:
Something's not very clear to me: the global value: `global.keycloak.postgresPassword.value` and the secret `keycloak-password-secret` are a redundant: I don't understand if both should be present, or either depending if the operators is performing an upgrade or a clean installation.
Perhaps we could clarify in the instructions?

Reply:
The value of `global.keycloak.postgresPassword.value` is written into `keycloak-password-secret`. If `global.keycloak.postgresPassword.value` is missing, the actual password is autogenerated through helm. In this second case, we have to copy the password from the secret into the values file if we want the password to remain unchanged over the deletion of the secret and the subsequent `helm upgrade`.

Reply:
I agree that the secrets handling could be documented better in general, but I'm not sure the instructions on how to upgrade the renku helm chart version is the best place for this.

Reply:
ok, thanks for the explanation.